src/w3d/renderer/assetmgr.cpp

```diff
@@ -795,12 +847,40 @@ void GameAssetManager::Recolor_Vertex_Material(VertexMaterialClass *vmat, uint32
 // 0x00765A40
 bool GameAssetManager::Recolor_Mesh(RenderObjClass *robj, uint32_t colour)
 {
-    // Uses MeshClass requires MaterialInfoClass
-#ifdef GAME_DLL
-    return Call_Method<bool, W3DAssetManager, RenderObjClass *, uint32_t>(0x00765A40, this, robj, colour);
-#else
-    return false;
-#endif
+    bool recolored = false;
+    MeshClass *mesh = static_cast<MeshClass *>(robj);
+    MeshModelClass *model = mesh->Get_Model();
+    MaterialInfoClass *matinfo = mesh->Get_Material_Info();
+    const char *str = strchr(mesh->Get_Name(), '.');
+    const char *str2;
+
+    if (str && (str2 = str + 1, *str) || (str2 = mesh->Get_Name()) != 0) {

```

Comment:
This is a bit horrible. strchr returns the pointer to the '.' or nullptr so the *str check will always be true. If we remove that we can simplify it to the following:
```
str2 = str == nullptr ? str + 1 : mesh->Get_Name();
if (str2 != nullptr){
   ....
}
```

Reply:
fixed.